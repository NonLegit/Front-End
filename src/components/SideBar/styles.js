import { styled } from '@mui/system';

export const SideBarContainer = styled('div')(({ theme, responsive }) => ({

  [theme.breakpoints.between('0', '1000')]: {
    display: responsive ? 'none' : 'block',
  },

  width: 312,
  marginLeft: 24,
  position: 'relative',
}));
