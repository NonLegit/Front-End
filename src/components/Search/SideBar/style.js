import { styled } from '@mui/material/styles';

export const SideBarContainer = styled('div')(({ theme }) => ({
  [theme.breakpoints.between('0', '992')]: {
    display: 'none',
  },
  width: 312,
  height: 4000,
  display: 'flex',
  flexDirection: 'column',
  marginLeft: 24,
  backgroundColor: 'transparent',
  position: 'relative',
}));
export const CommunityContainer = styled('div')({
  background: 'white',
  borderRadius: 4,
  marginBottom: 16,
  maxWidth: '100%',
  backgroundColor: 'white',
});
