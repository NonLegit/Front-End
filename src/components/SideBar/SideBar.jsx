import { SideBarContainer } from './styles';
/**
 * This component works as a container for the right half components
 * like communities in home page
 *
 * @component SideBar
 * @property {React.Component} children -Components which will be wrapped in sidebar
 * @returns {React.Component} Container represents its children
 */

function SideBar({ children, responsive }) {
  return (
    <SideBarContainer responsive={responsive}>
      {children}
    </SideBarContainer>
  );
}

export default SideBar;
