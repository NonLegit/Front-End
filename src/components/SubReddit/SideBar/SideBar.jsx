import { useState, useEffect } from 'react';
import { SideBarContainer, CommunityContainer } from './style';
import BackHome from './BackHome/BackHome';
import About from './AboutSubReddit/About';
import Moderators from './Moderators/Moderators';
import Flirt from './Flirt/Flirt';
import Moderation from './Mderation/Moderation';
import numFormatter from '../../../utils/MembersNum';
import UserLogin from '../../../authentication';
import Rules from './Rules/Rules';
/**
 * SideBar for subreddit
 * @component
 * @property {string} disc - description data
 * @property {object} topics -list of community topics
 * @property {string} Name -name of the subreddit
 * @property {string} primaryTopic -primary topic for subreddit
 * @property {string} createdAt -date of creating the subreddit
 * @property {object} moderatoesName -list of object of moderators
 * @property {string} username -user name
 * @property {object} rules -list of object of rules
 * @property {Boolean} createPost -to detect the sidebar in subreddit or creat post page

 * @return {React.Component} - SideBar for subreddit
 */
function SideBar(props) {
  const {
    disc, topics, Name, primaryTopic, createdAt, moderatoesName, username, members, rules, createPost,
  } = props;
  console.log('create post flag', createPost);
  const [moderate, setModerate] = useState(false);
  const finalArray = moderatoesName?.map((obj) => obj.user.userName);
  // const finalArray = moderatoesName?.map((obj) => obj.userName);

  const num = numFormatter(members);
  const mode = UserLogin(finalArray);
  console.log(mode);
  useEffect(() => {
    if (mode) {
      setModerate(true);
    } else {
      setModerate(false);
    }
  }, [moderatoesName, username, mode]);
  return (
    <SideBarContainer createPost>
      {!moderate
      && (
      <CommunityContainer>
        <About disc={disc} createdAt={createdAt} num={num} Name={Name} createPost={createPost} />
      </CommunityContainer>
      )}
      {moderate && (
      <CommunityContainer>
        <Moderation topics={topics} disc={disc} Name={Name} primaryTopic={primaryTopic} createdAt={createdAt} num={num} createPost={createPost} />
      </CommunityContainer>
      )}
      { (rules?.length > 0)
     && (
     <CommunityContainer>
       <Rules Name={Name} rules={rules} />
     </CommunityContainer>
     )}
      <CommunityContainer>
        <Flirt />
      </CommunityContainer>
      <CommunityContainer sx={{ padding: '0px 12px' }}>
        <Moderators moderatoesName={finalArray} />
      </CommunityContainer>
      {!createPost && <BackHome sx={{ marginTop: -5 }} />}
    </SideBarContainer>
  );
}

export default SideBar;
