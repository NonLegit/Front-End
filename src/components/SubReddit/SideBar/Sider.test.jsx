import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import numFormatter from '../../../utils/MembersNum';
import SideBar from './SideBar';

// test snapshot
test('test snapshot', async () => {
  const props = {
    Name: 'aaaa',
    topics: [],
    primaryTopic: 'aaaa',
    createdAt: '',
    moderatoesName: [],
    username: 'aaa',
    members: 500,
    rules: [],
    createPost: false,
    disc: 'aaaaaa',
  };
  const tree = renderer.create(
    <Router>
      <SideBar {...props} />
    </Router>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

// test Number Formatter
describe(' test Number Formatter', () => {
  const tests = [{
    num: 100000,
    output: '100.0K',
  },
  {
    num: 10000000,
    output: '10.0M',
  }, {
    num: 10,
    output: 10,
  }];
  tests.forEach((item, index) => {
    it(`test case ${index}`, () => {
      const result = numFormatter(item.num);
      // console.log('nour basma eslam madbouly ', result);
      expect(result).toEqual(item.output);
    });
  });
});
