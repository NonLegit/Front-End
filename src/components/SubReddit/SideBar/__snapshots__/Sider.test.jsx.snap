// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="css-15rkw9o"
>
  <div
    className="css-1868w9n"
  >
    <div
      className="css-l1ctgm"
    >
      <h2
        className="css-dgde1h"
      >
        About Community
      </h2>
      <button
        className="css-t5krk3"
      >
        <div
          className="MuiBox-root css-uwcr2l"
        >
          <div
            className="MuiBox-root css-k008qs"
            data-testid="sort"
            onClick={[Function]}
            onTouchEnd={[Function]}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MoreHorizIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
          </div>
        </div>
      </button>
    </div>
    <h2
      className="css-2v81eu"
    >
      <h2
        className="css-yevfdk"
      >
        aaaaaa
      </h2>
    </h2>
    <div
      className="css-1maxkpg"
    >
      <div
        className="css-knd9j4"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="EmailOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6zm-2 0-8 5-8-5h16zm0 12H4V8l8 5 8-5v10z"
          />
        </svg>
      </div>
      <h2
        className="css-1nrunep"
      >
        Created
         
        Invalid date
      </h2>
    </div>
    <hr
      className="css-1m0ad5p"
    />
    <div
      className="MuiBox-root css-10ucsqm"
    >
      <span>
        <div
          className="css-p1puod"
        >
          500
        </div>
        <p
          className="css-1cgnkz"
        >
          Members
        </p>
      </span>
      <span>
        <div
          className="css-1o97w51"
        >
          1.0k
        </div>
        <p
          className="css-1cgnkz"
        >
          Online
        </p>
      </span>
      <span>
        <div
          className="css-p1puod"
        >
          Top 1%
        </div>
        <p
          className="css-1cgnkz"
        >
          Ranked by Size
        </p>
      </span>
      <div />
    </div>
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-jf1sgt-MuiTypography-root-MuiLink-root"
      onBlur={[Function]}
      onFocus={[Function]}
      to="/submit/r/aaaa"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1anzha9-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        fontSize={15}
        fontWeight="bold"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        padding="4px"
        tabIndex={0}
        type="button"
      >
        create Post
      </button>
    </a>
  </div>
  <div
    className="css-1868w9n"
  >
    <div />
  </div>
  <div
    className="css-1o6gy2p"
  >
    <div
      className="css-1f2ln3t"
    >
      <h2
        className="css-dgde1h"
      >
        Moderators
      </h2>
    </div>
    <a
      className="css-vgri8"
      href="/"
      onClick={[Function]}
    >
      <button
        className="MuiButtonBase-root css-azyh4b-MuiButtonBase-root"
        disabled={false}
        fontSize={15}
        fontWeight="bold"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        padding="4px"
        tabIndex={0}
        type="button"
        variant="outlined"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ubwraf-MuiSvgIcon-root"
          data-testid="MailOutlineIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
          />
        </svg>
         
        Message the Mods
      </button>
    </a>
    <div
      className="css-e3tcub"
    >
      <a
        className="css-qq790j"
        href="/"
        onClick={[Function]}
      >
        VIEW ALL MODERATORS
      </a>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-ub553u-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    fontSize={15}
    fontWeight="bold"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Back to Top
  </button>
</div>
`;
